src/boot/loader.test.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Zextras <https://www.zextras.com>
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import { rest } from 'msw';
+import { act, screen } from '@testing-library/react';
+import React from 'react';
+import server from '../mocks/server';
+import { GetComponentsJsonResponseBody } from '../mocks/handlers/components';
+import { setup } from '../test/utils';
+import { Loader } from './loader';
+import { LOGIN_V3_CONFIG_PATH } from '../constants';
+
+jest.mock('../workers');
+jest.mock('../reporting/functions');
+
+describe('Loader', () => {
+	test('If only getComponents request fails, the LoaderFailureModal appears', async () => {
+		// using getInfo and loginConfig default handlers
+		server.use(
+			rest.get<never, never, GetComponentsJsonResponseBody>(
+				'/static/iris/components.json',
+				(req, res, ctx) => res(ctx.status(503))
```

Comment:
The test is failing only because there is no valid json in the response. I think we should improve how the fetch is made here
https://github.com/Zextras/carbonio-shell-ui/blob/ef3a88824e75db180b7e26a0e3d872387730078a/src/network/get-components.ts#L12-L14
and check also the status of the response.
See also here https://developer.mozilla.org/en-US/docs/Web/API/fetch, the fetch does not reject if a 404 is returned, but I think we should make the promise throw in this case too.

So, here, the handler could become
```suggestion
			rest.get<never, never, Partial<GetComponentsJsonResponseBody>>(
				'/static/iris/components.json',
				(req, res, ctx) => res(ctx.status(503, 'Controlled error: fail components.json request'), ctx.json({}))
```

while the fetch could be
```ts
fetch('/static/iris/components.json')
    .then((response) => {
        if (response.ok) {
            return response.json();
        }
        throw Error(response.statusText);
    })
```
And inside the jest-env-setup, this line can be removed
https://github.com/Zextras/carbonio-shell-ui/blob/ef3a88824e75db180b7e26a0e3d872387730078a/src/jest-env-setup.ts#L30